airbyte-featureflag/src/main/kotlin/Context.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.featureflag
+
+/**
+ * Context abstraction around LaunchDarkly v6 context idea
+ *
+ * I'm still playing around with this.  Basically the idea is to define our own custom context types
+ * (by implementing this sealed interface) to ensure that we are consistently using the same identifiers
+ * throughout the code.
+ *
+ * @property [kind] determines the kind of context the implementation is,
+ * must be consistent for each type and should not be set by the caller of a context
+ * @property [key] is the unique identifier for the specific context, e.g. a user-id or workspace-id
+ */
+sealed interface Context {
+    val kind: String
```

Comment:
why make this key a string as opposed to an enum? it seems like that's how it's functionally being used anyway.

Reply:
It's a string primarily for two reasons:
1.  LaunchDarkly requires a string to be provided in [their v6 api](https://github.com/airbytehq/airbyte/blob/9cb32d3381a12ba55f58379d4ddeeaccd2343984/airbyte-featureflag/src/main/kotlin/Client.kt#L187), so making it an Enum adds one more layer of abstraction without really giving us anything in return
2. As this is a `sealed interface` the`kind` value can only be set by the implementations of this interface within the `airbyte-featureflag` module (that is to say, no implementation of this interface can be defined outside of this module), so there is no risk of these values being used anywhere else

They could be an enum, but I question what value that would add.